frontend/lib/views/profile.dart

```diff
@@ -0,0 +1,236 @@
+              ),
+              SizedBox(height: 16),
+              
+              // User Name
+              StyleText(
+                text: userName,  // Updated to use state variable
+                style: ThemeText.header.copyWith(fontSize: 24),
+              ),
+              
+              // Email
+              Text(
+                'aubreydrake@mail.utoronto.ca',  // TODO: Replace with actual email from backend
+                style: TextStyle(
+                  color: Colors.grey[600],
+                  fontSize: 16,
+                ),
+              ),
+              SizedBox(height: 24),
+              
+              // Stats Row
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: [
+                  // TODO: Replace with actual stats from backend
+                  _buildStatColumn('Listings', '12'),
+                  _buildStatColumn('Sold', '8'),
+                  _buildStatColumn('Rating', '4.5'),
+                ],
+              ),
+              
+              SizedBox(height: 24),
+              
+              // Action Buttons
+              _buildActionButton('My Listings', Icons.list_alt, () {
+                // TODO: Navigate to listings
+              }),
+              _buildActionButton('Saved Items', Icons.favorite_border, () {
+                // TODO: Navigate to saved items
+              }),
+              _buildActionButton('Purchase History', Icons.history, () {
+                // TODO: Navigate to purchase history
+              }),
+              _buildActionButton('Edit Profile', Icons.edit, () {
+                _showEditProfileDialog(context); // added this temporary dailog
+              }),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildStatColumn(String label, String value) {
+    return Column(
+      children: [
+        Text(
+          value,
+          style: TextStyle(
+            fontSize: 20,
+            fontWeight: FontWeight.bold,
+          ),
+        ),
+        Text(
+          label,
+          style: TextStyle(
+            color: Colors.grey[600],
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildActionButton(String label, IconData icon, VoidCallback onPressed) {
+    return Padding(
+      padding: EdgeInsets.symmetric(vertical: 8.0),
+      child: ElevatedButton(
+        onPressed: onPressed,
+        style: ElevatedButton.styleFrom(
+          minimumSize: Size(double.infinity, 50),
+          alignment: Alignment.centerLeft,
+          padding: EdgeInsets.symmetric(horizontal: 16),
+        ),
+        child: Row(
+          children: [
+            Icon(icon),
+            SizedBox(width: 16),
+            Text(label),
+            Spacer(),
+            Icon(Icons.chevron_right),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _showEditProfileDialog(BuildContext context) {
```

Comment:
Same comment to the effect of code nesting.

You should identify the sub-widgets and use helper build functions or `final` variables to help reduce nesting in the code.